File: 2024/programacao_avancada/2024.1/03-05-2024/chinelos.cpp
#include <bits/stdc++.h>

using namespace std;

#define all(x) x.begin(), x.end()
#define ll long long int
#define esac(x) cout << x << "\n";
const int INF = 0x3f3f3f3f;
const ll LINF = 0x3f3f3f3f3f3f3f3f;
const int MAX = 1e3+10;


int main(){
    ios_base::sync_with_stdio(0); cin.tie(NULL); cout.tie(NULL);
    int n;
    cin >> n;
    vector<int> a(n+1);
    for (int i = 1; i <= n; i++){
        cin >> a[i];
    }
    int p, aux, ans = 0;
    cin >> p;
    for (int i = 0; i < p; i++){
        cin >> aux;
        if (a[aux] > 0){
            ans++;
            a[aux]--;
        }
    }
    esac(ans);
    return 0;
}

File: 2024/introducao_programacao/2024.1/07-06-2024/chinelos.py
n = int(input())
a = [0] * (n+1)
for i in range(1, n+1):
    a[i] = int(input())
p = int(input())
ans = 0
for i in range(p):
    aux = int(input())
    if a[aux] > 0:
        ans += 1
        a[aux] -= 1
print(ans)